cpp/tests/join/semi_join_tests.cpp

```diff
@@ -31,6 +32,80 @@
 
 template <typename T>
 using column_wrapper = cudf::test::fixed_width_column_wrapper<T>;
+using strcol_wrapper = cudf::test::strings_column_wrapper;
+using CVector        = std::vector<std::unique_ptr<cudf::column>>;
+using Table          = cudf::table;
 
 struct JoinTest : public cudf::test::BaseFixture {
 };
+
+TEST_F(JoinTest, SemiJoinWithStructsAndNulls)
+{
+  column_wrapper<int32_t> col0_0{{3, 1, 2, 0, 2}};
+  strcol_wrapper col0_1({"s1", "s1", "s0", "s4", "s0"}, {1, 1, 0, 1, 1});
+  column_wrapper<int32_t> col0_2{{0, 1, 2, 4, 1}};
+  std::initializer_list<std::string> col0_names = {
+    "Samuel Vimes", "Carrot Ironfoundersson", "Detritus", "Samuel Vimes", "Angua von Überwald"};
+  auto col0_names_col = strcol_wrapper{col0_names.begin(), col0_names.end()};
+  auto col0_ages_col  = column_wrapper<int32_t>{{48, 27, 351, 31, 25}};
```

Comment:
Just a question here. Should these ages correlate with the names? I see Samuel Vimes is repeated in the names column.

Reply:
I'm sure that originally it all did, but I've been using this as just data and not actually tying the names and ages together.